src/ServerRequestFactoryInterface.php

```diff
@@ -12,20 +12,9 @@
      *
      * @param string $method
      * @param UriInterface|string $uri
+     * @param array $server optional server params ($_SERVER or similar structure)
      *
      * @return ServerRequestInterface
      */
-    public function createServerRequest($method, $uri);
-
-    /**
-     * Create a new server request from server variables.
-     *
-     * @param array $server Typically $_SERVER or similar structure.
-     *
-     * @return ServerRequestInterface
-     *
-     * @throws \InvalidArgumentException
-     *  If no valid method or URI can be determined.
-     */
-    public function createServerRequestFromArray(array $server);
+    public function createServerRequest($method, $uri, $server = []);
```

Comment:
This is [how the interface used to look](https://github.com/http-interop/http-factory/pull/23) (with a different order) and it was problematic because the method and URI can exist in both parameters and in `$server`. This makes it unclear which is the preferred "source of truth". It *could* be documented, but I feel that is less than ideal, because the method signature itself should be enough to make it clear. A combined custom/global signature never will.

My feeling is that this should be rejected. @http-interop/http-middleware-contributors ?